explainers.md

```diff
@@ -0,0 +1,860 @@
+
+Example:
+
+
+```
+manifest_url: https://example.com/abc/manifest.json
+start_url:../start.html 
+
+-> processed start_url: https://example.com/start.html
+```
+
+
+
+#### [scope](https://www.w3.org/TR/appmanifest/#scope-member)
+
+Scope defines the navigation scope of this web app. Sometimes a website wants to host multiple apps with the same origin or within the same path on an origin. Eg: `https://example.com/chat`, `https://example.com/doc`, `https://example.com/path/app1`, `https://example.com/path/app2`.
+
+scope is **processed **by user agent to be a full URL as:
+
+
+
+*   The default is the start\_url’s base URL.
+*   If scope is “.”, the default will also be used.  
+*   If the scope is a relative URL, manifest\_url is used as base URL.
+*   If the manifest is hosted under another origin, the scope has to be specified as absolute URL with the same origin as the document URL.
+
+Example:
+
+
+```
+start_url: http://example.com/start/abc
+scope: ""
+-> processed scope: https://example.com/start
+
+manifest_url: https://example.com/manifest/manifest.json
+scope: "../"
+-> processed scope: https://example.com/
+```
+
+
+
+#### updatable
+
+Able to update certain parts of the web app’s metadata and still be considered the **same app**.
+
+
+#### global_id
+
+Id that guarantees global uniqueness, will be used by external PWA stores. Could be a composite key. This document will be exploring options for the global\_id used to identify apps, as well as the default global\_id to be used when it’s not specified.
+
+
+#### migration
+
+Backend of the user agent needs to internally migrate data. See the 
+[migration](#migration-1) section below for details here. This means adding implementation and operational costs for the specific user agent.
+
+On desktop, this means both Firefox and Chromium. Desktop Chromium also supports syncing apps across users’ devices, migration can be done for devices that update to the latest version of Chrome, but old versions will get synced with two duplicate apps.
+
+
+#### processed field
+
+A **processed** field, eg: processed start\_url is the returned full URL when user agents follow [W3 spec](https://www.w3.org/TR/appmanifest/#start_url-member) to parse the manifest. The processed field is saved in the user agent's db for this web app entity.
+
+Example:
+
+
+```
+Document at https://example.com/home/
+with <link rel="manifest" href='manifest.json' />
+Manifest:
+{
+  …
+  start_url: "../start",
+  scope: ".",
+  …
+}
+->
+processed manifest_url: https://example.com/home/manifest.json
+processed start_url: https://example.com/start
+processed scope: https://example.com/
+```
+
+
+
+#### Android implementations
+
+When discussing behaviors on Android, this doc is exclusively only covering Chromium-based browsers on Android.
+
+
+#### Desktop implementations
+
+When discussing behaviors on desktop, this doc covers Chromium-based browsers and Firefox (desktop).
+
```

Comment:
It might be worth adding iOS implementation here too. Asking around now to see what that is.